src/ui.py

```diff
@@ -144,60 +149,169 @@ def on_cancel(self):
+
+    def on_cancel(self):
+        self.destroy()
+
 
 class DataPanel(SubPage):
     def __init__(self, master, switch_page_callback):
         super().__init__(master, switch_page_callback)
+        self.current_index = -1
         self.search_frame = ttk.Frame(self)
-        self.search_frame.pack(side=bconst.TOP,fill=bconst.X)
+        self.search_frame.pack(side=bconst.TOP, fill=bconst.X)
         self.search_var = ttk.StringVar(value="")
-        btn = create_button("search",self.search)
-        self.create_form_entry(self.search_frame,"Search",self.search_var,anchor="w",buttons=[btn])
+        btn = create_button("search", self.search)
+        self.create_form_entry(self.search_frame, "Search", self.search_var, anchor="w", buttons=[btn])
 
         self.common_frame = ttk.Frame(self)
-        self.common_frame.pack(side=bconst.TOP,fill=bconst.X)
+        self.common_frame.pack(side=bconst.TOP, fill=bconst.X)
         self.button_frame = ScrolledFrame(self.common_frame, autohide=True)
-        self.button_frame.pack(side=bconst.LEFT,padx=5, fill=bconst.BOTH)
+        self.button_frame.pack(side=bconst.LEFT, padx=5, fill=bconst.BOTH)
 
         self.details_frame = ttk.Frame(self.common_frame)
-        self.details_frame.pack(side=bconst.LEFT,padx=30)
+        self.details_frame.pack(side=bconst.LEFT, padx=30)
         anchor = "w"
 
         self.entry_description_var = ttk.StringVar(value="")
         copy_desc_btn = create_button(
             "copy",
-            lambda:self.copy_to_clipboard(self.entry_description_var.get())
-            )
+            lambda: self.copy_to_clipboard(self.entry_description_var.get())
+        )
         self.create_form_entry(
             self.details_frame,
             "Description",
             self.entry_description_var,
             buttons=[copy_desc_btn],
             anchor=anchor,
-            )
+        )
 
         self.entry_username_var = ttk.StringVar(value="")
         copy_uname_btn = create_button(
             "copy",
-            lambda:self.copy_to_clipboard(self.entry_username_var.get())
-            )
+            lambda: self.copy_to_clipboard(self.entry_username_var.get())
+        )
         self.create_form_entry(
             self.details_frame,
-            "Username",self.entry_username_var,
+            "Username", self.entry_username_var,
             buttons=[copy_uname_btn],
             anchor=anchor,
-            )
+        )
 
         self.entry_password_var = ttk.StringVar(value="")
         copy_passw_btn = create_button(
             "copy",
-            lambda:self.copy_to_clipboard(self.entry_password_var.get())
+            lambda: self.copy_to_clipboard(self.entry_password_var.get())
         )
         toggle_show_passw_btn = create_button(
             "show",
-            lambda:self.toggle_show_entry_password()
-            )
+            lambda: self.toggle_show_entry_password()
+        )
+        toggle_update_passw_btn = create_button(
```

Comment:
This is not a toggle button. It is an update button. Same as the method it is doing is not toggling but updating